lib/class-wp-rest-image-editor-controller.php

```diff
@@ -99,99 +158,16 @@ public function permission_callback( $request ) {
-			return new WP_Error( 'rest_cannot_find_attached_file', __( 'Unable to find original media file.', 'gutenberg' ), array( 'status' => 500 ) );
-		}
-
-		$image_editor = wp_get_image_editor( $image_path );
-		if ( ! $image_editor->load() ) {
-			return new WP_Error( 'rest_cannot_load_editor', __( 'Unable to load original media file.', 'gutenberg' ), array( 'status' => 500 ) );
-		}
-
-		$size = $image_editor->get_size();
-
-		// Finally apply the modifications.
-		$crop_x = round( ( $size['width'] * floatval( $params['x'] ) ) / 100.0 );
-		$crop_y = round( ( $size['height'] * floatval( $params['y'] ) ) / 100.0 );
-		$width  = round( ( $size['width'] * floatval( $params['width'] ) ) / 100.0 );
-		$height = round( ( $size['height'] * floatval( $params['height'] ) ) / 100.0 );
-		$image_editor->crop( $crop_x, $crop_y, $width, $height );
-
-		if ( isset( $params['rotation'] ) ) {
-			$image_editor->rotate( 0 - $params['rotation'] );
-		}
-
-		// TODO: Generate filename based on edits.
-		$target_file = 'edited-' . $meta['original_name'];
-
-		$filename = rtrim( dirname( $image_path ), '/' ) . '/' . $target_file;
-
-		// Save to disk.
-		$saved = $image_editor->save( $filename );
-
-		if ( is_wp_error( $saved ) ) {
-			return $saved;
-		}
-
-		// Update attachment details.
-		$attachment_post = array(
-			'guid'           => $saved['path'],
-			'post_mime_type' => $saved['mime-type'],
-			'post_title'     => pathinfo( $target_file, PATHINFO_FILENAME ),
-			'post_content'   => '',
-			'post_status'    => 'inherit',
-		);
-
-		// Add this as an attachment.
-		$attachment_id = wp_insert_attachment( $attachment_post, $saved['path'], 0, true );
-
-		if ( is_wp_error( $attachment_id ) ) {
-			if ( 'db_update_error' === $attachment_id->get_error_code() ) {
-				$attachment_id->add_data( array( 'status' => 500 ) );
-			} else {
-				$attachment_id->add_data( array( 'status' => 400 ) );
+		$modifiers = array();
+		foreach ( $request['modifiers'] as $modifier ) {
+			if ( 'rotate' === $modifier['modifier'] ) {
+				$modifiers[] = new Image_Editor_Rotate( $modifier['angle'] );
+			} elseif ( 'flip' === $modifier['modifier'] ) {
+				$modifiers[] = new Image_Editor_Flip( $modifier['vertical'], $modifier['horizontal'] );
+			} elseif ( 'crop' === $modifier['modifier'] ) {
+				$modifiers[] = new Image_Editor_Crop( $modifier['left'], $modifier['top'], $modifier['width'], $modifier['height'] );
 			}
-
-			return $attachment_id;
-		}
-
-		// Generate thumbnails.
-		$metadata = wp_generate_attachment_metadata( $attachment_id, $saved['path'] );
-
-		$metadata['richimage'] = $meta;
-
-		wp_update_attachment_metadata( $attachment_id, $metadata );
-
-		$path     = '/wp/v2/media/' . $attachment_id;
-		$response = rest_do_request( $path );
-
-		if ( ! $response->is_error() ) {
-			$response->set_status( 201 );
-			$response->header( 'Location', rest_url( $path ) );
 		}
-
-		return $response;
+		return $this->editor->modify_image( $request['media_id'], $modifiers );
```

Comment:
```suggestion
		$response = $this->editor->modify_image( $request['media_id'], $modifiers );
		if ( is_wp_error( $response ) ) {
		     return $response;
		} 
		$response = rest_ensure_response( $response );

		return $response;
```

Make sure that response is valid.